File: src/main/scala/checker/Pop.java
package checker;

import java.util.Set;

public class Pop extends BranchOp{
    public Pop(Set<Integer>[] domains){super(domains);}
    @Override
    public BranchOp clone(){
        return new Pop(domains.clone());
    }
    @Override
    public String toString() {
        return "Pop\n";
    }
}

File: src/main/scala/checker/incremental/Pop.scala
package checker.incremental

/**
  * branching operation representing a pop operation.
  * It means that the trail of the solver should be
  * popped.
  *
  * @param doms : 'should be' values of the domains
  *             before the pop branching operation.
  */
class Pop(val doms: Array[Set[Int]]) extends BranchOp(doms) {
  override def clone = new Pop(domains.clone)

  override def toString = "Pop"
}